### ISSUE TITLE
Deterministic traits in units can occur at scenario start

### ISSUE THREAD

**Issue by sigurdfdragon** [wesnoth/wesnoth#1818]
Units placed in a [side] without traits specified can end up have the same traits each time the start of scenario save is loaded.

To reproduce:
Start THoT, use debug:cl to jump ahead to 11_The_Court_of_Karrag
Karrag's guards in the castle on the right will have the same traits every time the start of scenario is loaded, though they can be different depending on the steps taken to create the start of scenario save. But once it is created, the same traits will always be applied, even though they are supposed to be random.

**Comment by gfgtdf** [comment]
yes as seen here https://github.com/wesnoth/wesnoth/blob/2fe72820d14c75027b032ea1f61a84f684634981/src/game_state.cpp#L207 the random eneratetor is explcitiyl st to be deterministic in this case. This is related to mp Synchronisation. Usually all random stuff happens with seeds that are received from the server but before the start event the usual way to ask the server for a seed might not work

**Comment by gfgtdf** [comment]
In particulat it would be nice if this could be fixed for multiplayer too (currently the host decided the rnadom seed and thus the random initial traits for units, and it'd be icer to use a seed form the server). even though it's not that important for mp since mp maps usually dont have units with rnadom traits at start.

**Comment by sevu** [comment]
I noted it in the wiki page for [side].

**Comment by gfgtdf** [comment]
> I noted it in the wiki page for [side].

I don't think this is important enouth to be mentioned there tbh since it doesn't really effect game play (except the 'you cannot save-load to get different trait' ), also we might change it an anytime.

**Comment by sevu** [comment]
And it's gone …
